services/wfm/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java

```diff
@@ -271,6 +296,24 @@ private boolean filterForwardPrimaryPath(FlowPath path) {
                 );
     }
 
+    /**
+     * Returns map with affected inactive flow and flow pathId set for rerouting.
+     */
+    public Map<Flow, Set<PathId>> getAffectedInactiveFlowsForRerouting(SwitchId switchId) {
+        log.info("Get affected inactive flows for switch {}", switchId);
+        return flowRepository.findDownFlows().stream()
+                .filter(f -> f.getPaths().stream()
```

Comment:
I would prefer to extend repository with required method and not fetching all flows here.
P.S. it's fine to filter out flows by segments since we do not interested in single switch flows here(kilda do not attempt to re-route them).

Reply:
Also there are no unit-tests

Reply:
Reworked and added tests.